### DISCUSSION TITLE
Datatable adding edit options

### DISCUSSION THREAD

**Discussion by AleHS-99** [flet-dev/flet#3863] [Q&A]
When I cycle through the rows of my datatable I want to add an iconbutton referring to the edit option, I send this option the id corresponding to the data it displays, but it always keeps the last id passed instead of keep the one that corresponds to you as they do the rest of the data
![image](https://github.com/flet-dev/flet/assets/84586035/8d01d238-55e1-4d08-95fc-7bb778d971b8)

**Flet version (`pip show flet`)**:
0.8.1

**Comment by ndonkoHenri** [comment]
What you can do is:
- add `data=ids` to that IconButton
- and instead of passing ids to `self.edit_notaid` instead pass `x.control.data`

This should be sufficient and produce the expected result. 
Let me know!
